### PR TITLE
Add example executable for ScalarTensor

### PR DESCRIPTION
**Pull request by guilara**
## Proposed changes

Given that ScalarTensor uses the generalized harmonic system, we add an executable that closely follows the structure of that system. We include an example Input file (Kerr BH + scalar wave) to run it.

This PR needs:
- A fix for the parsing of observables with the same name (e.g. `gh::Tags::Phi` and `CurvedScalarWave::Tags::Phi`) (#5329 )

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsdeppe** [comment]
@guilara would you mind rebasing this with all the recent changes on develop, just to make sure it works?

**Comment by guilara** [comment]
> @guilara would you mind rebasing this with all the recent changes on develop, just to make sure it works?

Will do

**Comment by guilara** [comment]
@nilsdeppe I have rebased the code. Seems to compile and run, just changed a few includes.

File: tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    InnerRadius: &InnerRadius 1.9
+    OuterRadius: 25.0
+    Interior:
+      ExciseWithBoundaryCondition:
+        ProductDirichletAnalyticAndAnalyticConstant:
+          GeneralizedHarmonicDirichletAnalytic:
+            AnalyticPrescription: *InitialData
+          ScalarAnalyticConstant:
+            Amplitude: 0.0
+    InitialRefinement: [0, 0, 4]
```
Comment (nilsdeppe) [inline tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml]: Would it be reasonable to reduce this to say lev 2 to reduce test time?

File: tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  - Trigger:
+      TimeCompares:
+        Comparison: GreaterThan
+        Value: 0.1
+    Events:
+      - Completion
+  - Trigger:
+      Slabs:
+        Specified:
+          Values: [10]
```
Comment (nilsdeppe) [inline tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml]: Can you decrease this to 5 and still have the test be good?

File: tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+          - Name: PointwiseL2Norm(GaugeConstraint)
+            NormType: L2Norm
+            Components: Sum
+          - Name: PointwiseL2Norm(ThreeIndexConstraint)
+            NormType: L2Norm
+            Components: Sum
+  - Trigger:
+      Slabs:
+        EvenlySpaced:
+          Interval: 1
```
Comment (nilsdeppe) [inline tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml]: Dumping every timestep seems a bit aggressive for volume data. Can this be reduced in frequency?

File: tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+      - ObserveFields:
+          SubfileName: VolumeData
+          VariablesToObserve:
+            - Csw(Psi)
+            - Lapse
+            - PointwiseL2Norm(GaugeConstraint)
+            - PointwiseL2Norm(ThreeIndexConstraint)
+          InterpolateToMesh: None
+          CoordinatesFloatingPointType: Double
+          FloatingPointTypes: [Double]
```
Comment (nilsdeppe) [inline tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml]: `Float` and above. There's rarely a reason to need to know the output to 16 digits for data visualization. This is an easy way to reduce output file size by 2

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Looks good, some suggestions in the input file for reducing runtime and the amount of data being dumped. Please squash immediately :)

File: tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml [hunk 5]
```diff
@@ -0,0 +1,255 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+Executable: EvolveScalarTensorSingleBlackHole
+Testing:
+  Check: parse;execute_check_output
+  Timeout: 8
```
Comment (nilsdeppe) [inline tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml]: Looks like this is still timing out, maybe bump to 20 seconds?

Comment (guilara) [reply tests/InputFiles/ScalarTensor/KerrSchildSphericalHarmonic.yaml]: I bumped to 20s, but then needed to also lower the output freq again and domain parameters.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**